### DISCUSSION TITLE
Cores with independent PWM frequencies

### DISCUSSION THREAD

**Discussion by Uttam1** [earlephilhower/arduino-pico#2077] [General]
Hi Earl,

Thank you for the amazing work. Would it be possible to run the two independent cores at their own PWM frequencies? Currently, with analogWriteFrequency it sets the frequency on both cores and a subsequent call to it in the second core has no effect. 

Thank you 
Uttam

**Comment by earlephilhower** [comment]
The PWM module has 2 slices, not connected to any specific cores, each of which can have a different clock divider.  Only one slice can talk to any GPIO pin and it's a very idiosyncratic configuration of duplicated A/B channels.

It's gonna be complicated, but you can use the raw SDK calls to configure the slices independently using the [RP2040 datasheet](https://datasheets.raspberrypi.com/rp2040/rp2040-datasheet.pdf) as a reference.  The Arduino core itself isn't going to support this since there's no good way of fitting that oddity into the Arduino way.
